utils/hare_cov/hare_coverage.py

```diff
@@ -0,0 +1,355 @@
+        :param pattern: Directory name to scan
+        :param callback: callback for comparison
+        :param kwargs: Keyword arguments for the filtering.
+        """
+        self.pattern = pattern
+        self.callback = callback or self.comparator
+        self.omit_kwargs = kwargs.get("omit", "py3venv build").split()
+
+    def comparator(self, dir_name: str) -> bool:
+        """
+        Compares directory name with a set pattern.
+        :param dir_name: directory name
+        :return: boolean
+        """
+        return dir_name == self.pattern
+
+    def omit_dir(self, dir_name: str) -> bool:
+        """
+        Filter out the omitted directories.
+        :param dir_name: directory name
+        :return: boolean
+        """
+        for word in self.omit_kwargs:
+            if dir_name.find(word) != -1:
+                return True
+        return False
+
+    def search_dir(self, dir_path: str) -> List[str]:
+        """
+        Strategy for scanning directories.
+        :param dir_path: directory path
+        :return: List of the file match to given pattern
+        """
+        result_list: List[str] = []
+        for sub_dir, dirs, _ in walk(dir_path):
+            for directory in dirs:
+                directory_path: str = sub_dir + sep + directory
+                if self.omit_dir(directory_path):
+                    continue
+                result_list.extend([ele for ele in
+                                    self.search_dir(directory_path)
+                                    if ele not in result_list])
+                if self.callback(directory):
+                    if directory_path not in result_list:
+                        result_list.append(directory_path)
+        return result_list
+
+
+class Utility:
+    """Here is the utility class which has static utility helper methods."""
+
+    def run_cmd(self, command: str, delimiter: Optional[str] = None,
+                max_split: int = -1) -> None:
+        """
+        Execute command and return the result.
+        :param command: Command in string format
+        :param delimiter: delimiter (default is space)
+        :param max_split: maximum number split
+        :return: tuple(stdout, stderr)
+        """
+        logging.debug(f"Executing Command : {command!a}")
+        process: Popen[bytes] = Popen(
+            command.split(sep=delimiter, maxsplit=max_split),
+            stdout=PIPE,
+            stderr=PIPE)
+
+        out, err = process.communicate()
+        if process.returncode:
+            raise Exception(
+                f'Command {command!a} exited with error code '
+                f'{process.returncode}.\n'
+                f'Command output: {err.decode("utf-8")}')
+
+        logging.debug(f'Output:\n{out.decode("utf-8")}')
+
+    def is_dir_exist(self, dir_path: str) -> bool:
+        """
+        Check for the existence of the file or directory.
+        :param dir_path: file or directory path

```

Comment:
If this function also check for a file path then the name is not appropriate. If dir path is the expected argument then you need to fix the comment accordingly.

Reply:
It is working for both file and folders so add new name to the function.